lib/heidrun/mapping_tools/marc.rb

```diff
@@ -0,0 +1,204 @@
+      def datafield_els(r, tag)
+        select_field(r, 'datafield', tag)
+      end
+
+      ##
+      # Return the String value of the datafield with the given number (tag)
+      #
+      # An empty array is returned if the datafield does not exist.
+      #
+      # @param  r   [Krikri::XmlParser::Value] The record root element
+      # @param  tag [String] The tag, e.g. '001'
+      # @return     [Array] of String ([] if element does not exist)
+      def datafield_values(r, tag)
+        select_field(r, 'datafield', tag).map { |f| f.children.first.to_s }
+      end
+
+      ##
+      # Return the String value of the controlfield with the given number (tag)
+      #
+      # @param  r   [Krikri::XmlParser::Value] The record root element
+      # @param  tag [String]  The tag, e.g. '007'
+      # @return     [String]
+      # @raise      [NoElementError]  If the controlfield doesn't exist
+      def controlfield_value(r, tag)
+        select_field(r, 'controlfield', tag).first.children.first.to_s
+      rescue NoMethodError
+        raise NoElementError.new "No control field #{tag}"
+      end
+
+      ##
+      # Return an Element for the MARC leader
+      #
+      # @param   r [Krikri::XmlParser::Value] The record root element
+      # @return    [String]
+      # @raise     [NoElementError]  If there is no leader
+      def leader_value(r)
+        r.node.children.select { |n| n.name == 'leader' }
+                       .first.children.first.to_s
+      rescue NoMethodError
+        raise NoElementError.new "No MARC leader element"
+      end
+
+      ##
+      # Return the String value of the subfield element with the given code
+      #
+      # An empty array is returned if the subfield can not be found.
+      #
+      # @param  elements [Element] The elemenets, e.g. datafield
+      # @param  code     [String]  Code, i.e. its 'tag' attribute
+      # @return          [Array]  ([] if the subfield can not be found)
+      def subfield_values(elements, code)
+        elements.map do |e|
+          nodes = e.children.to_a.select(&subfield_code_condition(code))
+          !nodes.empty? ? nodes.first.children.first.to_s : nil
+        end.compact
+      end
+
+      ##
+      # Return an array of Strings for the values of all of the subfields in
+      # the given element that have a-z codes
+      #
+      # @param element [Element] The element, which is probably a datafield
+      # @return        [Array]   An array of String.  Empty if no subfields.
+      def all_subfield_values(elements)
+        elements.map do |el|
+          el.children.to_a
+            .select { |child| child.name == 'subfield' \
+                              && child[:code] =~ /^[a-z]$/ }
+            .map { |child| child.children.first.to_s }
+        end.flatten
+      end
+
+      ##
+      # Whether Control Field 007 indicates Film / Video
+      # @param s [String] Control Field 007
+      def film_video?(s)
+        %w(c d f o).include?(s[1])
```

Comment:
Yeah, that was an oversight on my part, because position `06` of the leader is (often) also `g` in this case: http://www.loc.gov/marc/bibliographic/bdleader.html

Either way, you're right. Good catch! :)